components/sup/src/main.rs

```diff
@@ -223,21 +229,97 @@ fn mgrcfg_from_matches(m: &ArgMatches) -> Result<ManagerConfig> {
         )?,
         ..Default::default()
     };
+
     if let Some(addr_str) = m.value_of("LISTEN_HTTP") {
         cfg.http_listen = http_gateway::ListenAddr::from_str(addr_str)?;
     }
+
     if let Some(addr_str) = m.value_of("LISTEN_CTL") {
         cfg.ctl_listen = SocketAddr::from_str(addr_str)?;
     }
+
     if let Some(watch_peer_file) = m.value_of("PEER_WATCH_FILE") {
         cfg.watch_peer_file = Some(String::from(watch_peer_file));
     }
+
     if let Some(events) = m.value_of("EVENTS") {
         cfg.eventsrv_group = ServiceGroup::from_str(events).ok();
     }
+
+    let kf = m.value_of("KEY_FILE");
+    let cf = m.value_of("CERT_FILE");
+
+    if kf.is_some() && cf.is_some() {
+        let kpb = PathBuf::from(kf.unwrap());
+        let cpb = PathBuf::from(cf.unwrap());
+
+        validate_key_file(kpb.as_path())?;
+        validate_cert_file(cpb.as_path())?;
+
+        cfg.tls_files = Some((kpb, cpb));
+    }
+
     Ok(cfg)
 }
 
+fn validate_key_file(key_path: &Path) -> Result<()> {
+    match File::open(key_path) {
+        Ok(f) => {
+            let key_file = &mut BufReader::new(f);
+
+            if let Ok(keys) = pemfile::rsa_private_keys(key_file) {
+                if keys.len() > 0 {
+                    Ok(())
+                } else {
+                    Err(sup_error!(Error::InvalidKeyFile(
+                        key_path.to_path_buf(),
+                        String::from("No private keys inside key file.")
+                    )))
+                }
+            } else {
+                Err(sup_error!(Error::InvalidKeyFile(
+                    key_path.to_path_buf(),
+                    String::from("No private keys inside key file.")
+                )))
+            }
+        }
+        Err(e) => Err(sup_error!(Error::InvalidKeyFile(
+            key_path.to_path_buf(),
+            e.to_string()
+        ))),
+    }
+}
+
+fn validate_cert_file(cert_path: &Path) -> Result<()> {
+    match File::open(cert_path) {
+        Ok(f) => {
+            let cert_file = &mut BufReader::new(f);
+            let cert_chain = match pemfile::certs(cert_file) {
+                Ok(c) => c,
+                Err(_) => {
+                    return Err(sup_error!(Error::InvalidCertFile(
+                        cert_path.to_path_buf(),
+                        String::new()
```

Comment:
I added the `String` there because I felt like there were nuances that could be helpful to users beyond just "invalid file".  For example, sometimes you might be given a file that contains the text `haha`, which is clearly an invalid cert file in the most literal sense.  Other times, you might have a valid cert file, but the permissions on it might make it unreadable.  I thought having the opportunity to inject additional information, like the text returned from the `std::io::Error` might be helpful.  I'll look at streamlining this.

Reply:
I totally agree there could be additional nuances. I was suggesting that those nuances could be codified as additional kinds of errors, rather than ad-hoc strings at the point where the errors were thrown.

Reply:
In my current refactoring of this that I haven't pushed yet, the different types of errors all bubble up individually, so `std::io::Error` is returned as such, etc.  I think that should take care of this.